tests/integration/ha/test_ha.py

```diff
@@ -197,6 +197,77 @@ async def test_kill_db_process_node_with_primary_shard(
     await assert_continuous_writes_consistency(c_writes)
 
 
+@pytest.mark.abort_on_fail
+async def test_freeze_db_process_node_with_primary_shard(
+    ops_test: OpsTest, c_writes: ContinuousWrites, c_balanced_writes_runner
+) -> None:
+    """Check cluster can self-heal + data indexed/read on process freeze on node with P_shard."""
+    app = (await app_name(ops_test)) or APP_NAME
+
+    units_ips = get_application_unit_ids_ips(ops_test, app)
+    leader_unit_ip = await get_leader_unit_ip(ops_test, app=app)
+
+    # find unit hosting the primary shard of the index "series-index"
+    shards = await get_shards_by_index(ops_test, leader_unit_ip, ContinuousWrites.INDEX_NAME)
+    first_unit_with_primary_shard = [shard.unit_id for shard in shards if shard.is_prim][0]
+
+    # Killing the only instance can be disastrous.
+    if len(ops_test.model.applications[app].units) < 2:
+        await ops_test.model.applications[app].add_unit(count=1)
+        await ops_test.model.wait_for_idle(
+            apps=[app],
+            status="active",
+            timeout=1000,
+            idle_period=IDLE_PERIOD,
+        )
+
+    # Freeze the opensearch process
+    opensearch_pid = await send_kill_signal_to_process(
+        ops_test, app, first_unit_with_primary_shard, signal="SIGSTOP"
+    )
+
```

Comment:
Might need to sleep for the re-election period here

Reply:
there is a sleep a couple of lines below this line

Reply:
that makes sense - perhaps adding a comment above that line that says:
```
# assert that writes are continuing, sleeping for 5 seconds should be pleanty of time for the reelection to occur since it happens in an order of miliseconds
```
would be useful